common/notify_stream/src/lib.rs

```diff
@@ -1,18 +1,18 @@
 extern crate notify;
 
 use futures::{stream, Stream};
-use notify::{DebouncedEvent, Error as NotifyError, Watcher as NotifyWatcher};
+use notify::event::{CreateKind, DataChange, ModifyKind, RemoveKind, RenameMode};
+use notify::{Config, ErrorKind, EventKind, Watcher as NotifyWatcher};
 use std::path::Path;
 use std::rc::Rc;
-use std::time::Duration;
 use time::OffsetDateTime;
 
 type PathId = std::path::PathBuf;
 
 #[cfg(target_os = "linux")]
 type OsWatcher = notify::INotifyWatcher;
 #[cfg(not(any(target_os = "linux")))]
-type OsWatcher = notify::PollWatcher;
+type OsWatcher = notify::RecommendedWatcher;
```

Comment:
Using the `RecommendedWatcher` here because initially less tests were failing with it. It also fixes the another Windows issue we've been tracking.l